CONTRIBUTING.rst

```diff
@@ -0,0 +1,33 @@
+Contributing to MNE-Python
+==========================
+
+MNE-Python is maintained by a community of scientists and research labs, and
+accepts contributions in the form of bug reports, fixes, feature additions, and
+documentation improvements (even just typo corrections). The best way to start
+contributing is by `opening an issue`_ on our GitHub page to tell us about
+errors you are encountering or to discuss your ideas for changes. For general
+troubleshooting, you can also write to the `MNE mailing list`_ or chat with
+developers on the `MNE gitter channel`_.
+
+Changes are typically made by `forking`_ the MNE-Python repository, making
+changes to your fork (usually by `cloning`_ it to your personal computer,
+making the changes, and then `pushing`_ the local changes up to your fork), and
+finally creating a `pull request`_ to incorporate your changes back into the
+shared "upstream" version of the codebase. The documentation section
+":ref:`dev-setup`" has more details on how best to configure your workflow for
+a smooth experience contributing to MNE-Python.
+
+Users and contributors to MNE-Python are expected to follow our `code of
+conduct`_.
+
+
+.. _`opening an issue`: https://github.com/mne-tools/mne-python/issues/new/choose
+.. _`MNE mailing list`: http://mail.nmr.mgh.harvard.edu/mailman/listinfo/mne_analysis
+.. _`MNE gitter channel`: https://gitter.im/mne-tools/mne-python
+
+.. _`forking`: https://help.github.com/en/articles/fork-a-repo
+.. _`cloning`: https://help.github.com/en/articles/cloning-a-repository
+.. _`pushing`: https://help.github.com/en/articles/pushing-to-a-remote
+.. _`pull request`: https://help.github.com/en/articles/creating-a-pull-request-from-a-fork
+
+.. _`code of conduct`: https://github.com/mne-tools/mne-python/blob/master/CODE_OF_CONDUCT.md
```

Comment:
NOTE: a few of these links are redundant with entries in one of the various `.inc` files; however, this CONTRIBUTING file in the repo root needs to stand alone as it will often be read as plain text without being rendered.